ui/src/treeDataProvider/AzureParentTreeItem.ts

```diff
@@ -13,18 +13,18 @@ import { GenericTreeItem } from './GenericTreeItem';
 import { IAzureParentTreeItemInternal } from './InternalInterfaces';
 import { loadingIconPath, loadMoreLabel } from './treeConstants';
 
-export abstract class AzureParentTreeItem<T = ISubscriptionRoot> extends AzureTreeItem<T> implements types.AzureParentTreeItem<T>, IAzureParentTreeItemInternal<T> {
+export abstract class AzureParentTreeItem<TRoot = ISubscriptionRoot> extends AzureTreeItem<TRoot> implements types.AzureParentTreeItem<TRoot>, IAzureParentTreeItemInternal<TRoot> {
```

Comment:
Is there a way we could create a generic TRoot class that includes the properties that are required?

Reply:
`ISubscriptionRoot` is already the 'generic' or 'default' case. For example, the Storage extension never had to even specify this type: https://github.com/Microsoft/vscode-azurestorage/pull/230

App Service and Functions had to specify the type, but only because I wanted to add `SiteClient` to `ISubscriptionRoot` so I didn't have to pass it down the whole tree.